test/test.js

```diff
@@ -32,18 +32,18 @@ test('call Intl.NumberFormat correctly', (t) => {
     this.format = () => ({ locale, opts })
   }
 
-  t.is(formatNumber(0, { maxSignificant: 2 }).opts.maximumSignificantDigits, 2)
-  t.is(formatNumber(0, { maxSignificantDigits: 2 }).opts.maximumSignificantDigits, 2)
-  t.is(formatNumber(0, { maximumSignificant: 2 }).opts.maximumSignificantDigits, 2)
-  t.is(formatNumber(0, { minSignificant: 2 }).opts.minimumSignificantDigits, 2)
-  t.is(formatNumber(0, { minSignificantDigits: 2 }).opts.minimumSignificantDigits, 2)
-  t.is(formatNumber(0, { minimumSignificant: 2 }).opts.minimumSignificantDigits, 2)
-  t.is(formatNumber(0, { maxInteger: 2 }).opts.maximumIntegerDigits, 2)
-
-  t.deepEqual(formatNumber(0).locale, ['en-US'])
-  t.deepEqual(formatNumber(0).opts, {
+  t.is(formatNumber(0, { maxSignificant: 3 }).opts.maximumSignificantDigits, 3)
+  t.is(formatNumber(0, { maxSignificantDigits: 3 }).opts.maximumSignificantDigits, 3)
+  t.is(formatNumber(0, { maximumSignificant: 3 }).opts.maximumSignificantDigits, 3)
+  t.is(formatNumber(0, { minSignificant: 3 }).opts.minimumSignificantDigits, 3)
+  t.is(formatNumber(0, { minSignificantDigits: 3 }).opts.minimumSignificantDigits, 3)
+  t.is(formatNumber(0, { minimumSignificant: 3 }).opts.minimumSignificantDigits, 3)
+  t.is(formatNumber(0, { maxInteger: 3 }).opts.maximumIntegerDigits, 3)
+
+  t.deepEqual(formatNumber(0, { locale: 'ka-GE' }).locale, ['ka-GE'])
+  t.deepEqual(formatNumber(0, { locale: 'ka-GE' }).opts, {
     nanZero: true,
-    locale: 'en-US',
+    locale: 'ka-GE',
```

Comment:
not sure why we need all these changes. If we don't, can we undo them? And can we add a test for memoization?

Reply:
@mvayngrib ish. We need these because the  `this.format = () => ({ locale, opts })` function overrides normal behavior and returns opts instead of formatted number. Since the hack doesn't affect memoized number formatters tests fail, this way I force to recreate the NumberFormat obj

Reply:
not obvious from github changes, i see now it's because of

https://github.com/ExodusMovement/format-num/blob/028028eafe8ee41908c6e00683a68821bd4b5ad1/test/test.js#L31-L33

Reply:
these tests kinda weird, i'm ok with change. Maybe better to add some meaningful tests as well

Reply:
I agree. What's the point of testing opts

Reply:
agree, the perf change is great, but the tests need more clarity

Reply:
@mvayngrib Is that nit or should we go ahead with merge?

Reply:
non-blocking, but please capture

Reply:
Captured: https://app.asana.com/0/1202170937152401/1206553661682635